File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String[] solution(String myStr) {
        String[] srr = myStr.split("a|b|c");
        List<String> ret = new ArrayList<>();
        for(String s : srr){
            if(s.isEmpty()) continue;
            ret.add(s);
        }
        if(ret.isEmpty()) return new String[]{"EMPTY"};
        return ret.stream().toArray(String[]::new);
    }
}

File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.py

def spplit(s):
    deli = ['a', 'b', 'c']
    temp = ''
    ret = []
    for c in s:
        if c in deli:
            if temp:
                ret.append(temp)
            temp = ''
            continue
        temp += c
    if temp:
        ret.append(temp)
    if not ret:
        ret.append("EMPTY")
    return ret
    

def solution(myStr):
    return spplit(myStr)